lib/src/util/withClass.tsx

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { ComponentProps, ReactElement } from 'react';
+import classnames from 'classnames';
+
+type ElementType<E extends keyof JSX.IntrinsicElements> = ReactElement<ComponentProps<E>, E>;
+type ComponentConstructorType<E extends keyof JSX.IntrinsicElements> = (p: ComponentProps<E>) => ElementType<E>;
+
+export default function withClass<E extends keyof JSX.IntrinsicElements>(
+  withClassName: string,
+  El: E
+): ComponentConstructorType<E> {
+  return function WithClassWrapper({ className, ...otherProps }: ComponentProps<E>): ElementType<E> {
+    // There should be a safer way but I can't figure it out; TS keeps thinking that
```

Comment:
The top answers to this question look promising, but I'm not sure if they'll help or not here. https://stackoverflow.com/questions/55969769/typing-a-dynamic-tag-in-react-with-typescript

Reply:
> The top answers to this question look promising, but I'm not sure if they'll help or not here.

Thanks that link helpfed me fumble around until I got it.  927d0d67